tests/gateway/test_utils.py

```diff
@@ -0,0 +1,132 @@
+import pytest
```

Comment:
Moved the added test to the correct module level utils test module. 
Also, just added some minor test coverage for the utilities.

Reply:
> Also, just added some minor test coverage for the utilities.

I'd add this later in a follow-up and leave a TODO comment. Now this PR is blocked by my comments that are unrelated to this PR's goal.